cmd/stellar-rpc/internal/config/options.go

```diff
@@ -666,3 +706,44 @@ func positive(option *Option) error {
 	}
 	return nil
 }
+
+func setForNetwork(cfg *Config, networkParams networkConfig) error {
+	if slices.Equal(cfg.HistoryArchiveURLs, networkParams.historyArchiveURLs) && cfg.NetworkPassphrase == networkParams.networkPassphrase {
+		return nil
+	}
+	if len(cfg.HistoryArchiveURLs) != 0 || cfg.NetworkPassphrase != "" {
+		return errors.New("both network and (HistoryArchiveURLs or NetworkPassphrase) specified (only one may be chosen)")
```

Comment:
I'm really not a fan of the phrasing of the error but it's hard to figure out a better way to say it without ambiguity... what about:

```
network option conflicts with values provided for passphrase and/or history archive URLs
```